### PR TITLE
Return file version

### PR DESCRIPTION
**Pull request by alexbrodersen**
The file version_id can be useful for tracking versions but is not available in box_previous_versions for the currently "active" version of the file. This pull request updates box_ls() and the s3 as.data.frame method to include the version_id, as well as alters the name of the value of the current "version" to "version_no".

### CODE REVIEW

**Comment by nathancday** [comment]
Thanks for the PR. I like you idea of mirroring the behavior of `box_previous_versions` and the proposed name changes sound good.

Do you have a specific instance were you want to use version_id instead of version_no?

I've been working on adding a function, `box_file_version()` that uses file IDs to do something like this with version_no results. As part of that I've have been re-considering what we return our of `box_previous_versions()` and my thinking was that version_no alone was enough, but maybe not :)

**Comment by nathancday** [comment]
Got it. Paired with a file_id, either a version_id or version_no is enough. Numbers are inferred, but the Box web-app does that too. I've personally found it humanly easier to work with version_no, but I like your idea of having both easily available to users.

**Comment by alexbrodersen** [comment]
I agree, especially if using both boxr and Web App, the version_no is more straightforward to work with.

I do think both should be provided.

One case where the version_id matters is (at least in the enterprise version) is the URL for file version.

This has the form

https://{company_name}.app.box.com/file/{file_id}?sb=/activity/versions/{version_id}

and as far as I know version_no can't be used in this way.

**Comment by ijlyttle** [comment]
Let me put my own spin on `version_no`: this is a construct only of the {boxr} package, and is completely independent of the Box API, which knows only `file_version_id`. In other words, I agree that both are important.

We are currently working on a successor to `box_previous_versions()`, called `box_version_history()`.

One of the things I might like to implement in `box_version_history()` is to return a column `version_no`, rather than a character column ,`version` with values `"V1"`, `"V2"`, etc. - so this sounds like a good fit.  One of the neat things about introducing a new name is that we can get away with changing the API :) 

I think we should keep the `version` variable in the `as.data.frame()` method because this could be a breaking change for someone otherwise. I don't know how to deprecate a variable in a return object, maybe the least-bad thing would be to keep them both, awkward as that may be. Thoughts?

In short, I think that it would be good to standardize on the use of `version_no` within {boxr}, and this is a step in that direction.

There's another PR that I want to get wrapped-up first, then we may have some specific suggestions for this one.

**Comment by ijlyttle** [comment]
Another question - the use of `x$etag` predates my involvement. I have read around a bit ([google search](https://www.google.com/search?q=boxr+api+etag)), and I have not found anything that suggests that this is a rock-solid proxy for version number.

Of course, it was done this way for a reason - I would like to feel good about the reason while we have the question on the floor.

**Comment by ijlyttle** [comment]
Finally (for now), welcome @alexbrodersen!

So that you can feel fully a part of the fun: I had a quick look through our code, and see that we use refer to `file_version_id` and `version_id`, although Box uses `file_version_id`.

As a first thought, it might be easiest for {boxr} to standardize on `version_id`.

**Comment by alexbrodersen** [comment]
@ijlyttle Regarding depreciating a column in a return variable, I suppose this may represent a breaking API change, and boxr should increment version number if it's completely removed?

I do think the easiest solution, for now, is to leave that as is and extend the return with the new variables. 

Regarding `file_version_id` vs `version_id`, I'm in favor of keeping things relatively consistent with the box API, but in my opinion it's more important to be internally consistent across functions in the boxr package.

**Comment by alexbrodersen** [comment]
Here's what I get from grep for the list of where version_id vs file_version_id is used. I can update my PR to change this based on whatever is decided?

```
boxr__internal_get.R:63:      version_id <- versions$file_version_id[version_no]
boxr_file_versions.R:9:#' The returned `data.frame` contains a variable, `file_version_id`, 
boxr_file_versions.R:63:  colnames(d)[colnames(d) == "id"] <- "file_version_id"

boxr__internal_get.R:30:boxGet <- function(file_id, local_file, version_id = NULL, version_no = NULL,
boxr__internal_get.R:36:  if (!is.null(version_id) & !is.null(version_no)) {
boxr__internal_get.R:38:    stop("Only one of version_id and version_no may be supplied")
boxr__internal_get.R:41:  if (is.null(version_id) & !is.null(version_no)) {
boxr__internal_get.R:63:      version_id <- versions$file_version_id[version_no]
boxr__internal_get.R:67:  if (!is.null(version_id)) {
boxr__internal_get.R:68:    # If you have a version_id, check that it looks reasonable (11 digits)
boxr__internal_get.R:69:    version_id <- as.numeric(version_id)
boxr__internal_get.R:71:    if (!grepl("[[:digit:]]{11}", version_id))
boxr__internal_get.R:72:      stop("version_id must be an integer, 11 characters in length")
boxr__internal_get.R:79:        file_id, "/content", "?version=", version_id
boxr_file_versions.R:9:#' The returned `data.frame` contains a variable, `file_version_id`, 
boxr_file_versions.R:63:  colnames(d)[colnames(d) == "id"] <- "file_version_id"
boxr_read.R:56:box_read <- function(file_id, type = NULL, version_id = NULL, 
boxr_read.R:64:  req <- boxGet(file_id, local_file = temp_file, version_id = version_id, 
boxr_s3_classes.R:143:      version_id          = x$file_version$id %|0|% NA_character_,
boxr_upload_download.R:13:#' **`version_id`** or **`version_no`**.
boxr_upload_download.R:17:#' using the `version_id` parameter.
boxr_upload_download.R:24:#' [box_previous_versions()] to retrieve the `version_id`,
boxr_upload_download.R:32:#' @param version_id `character` or `numeric`, the `version_id` of the file.
boxr_upload_download.R:57:                   file_name = NULL, version_id = NULL, version_no = NULL,
boxr_upload_download.R:88:  req <- boxGet(file_id = file_id, version_id = version_id,
```

**Comment by ijlyttle** [comment]
Hi @alexbrodersen,

I think we are on the same page here.

If I were starting the package from scratch, I might have chosen `file_version_id` to be better aligned with the Box API, but I think the right move is to stick with `version_id` here.

We do plan to change the version number, but I think it would be "rude" for the `version` variable to just disappear - I think we can just have both for now, and think about how to remove it down the road.

The PR is in good shape now - I plan to get #159 taken care of immediately, then come back with specific suggestions for this PR - none of which will be very big or surprising.

**Comment by nathancday** [comment]
IMO if we keep `version` named as is and add `version_id` this is good to go. Maybe add a test for names exceptions.

The box web app also must be doing a similar version inference because it displays a v1, v2, v3, ... badge next to file names and in the version menu. I think we should continue to support both (human readable number and full ID) like `box_dl()` whenever possible. No reason `box_version` can't report both.

Screen Shot 2020-09-23 at 5 32 06 PM
Screen Shot 2020-09-23 at 5 32 27 PM

**Comment by ijlyttle** [comment]
Sounds good to me! I can provide some specific feedback here shortly.

File: R/boxr_s3_classes.R [hunk 1]
```diff
@@ -139,7 +139,8 @@ as.data.frame.boxr_object_list <- function(x, ...) {
       modified_at         = box_datetime(x$modified_at),
       content_modified_at = box_datetime(x$content_modified_at) %|0|% as.POSIXct(NA),
       sha1                = x$sha1 %|0|% NA_character_,
-      version             = as.numeric(x$etag) + 1,
```
Comment (ijlyttle) [inline R/boxr_s3_classes.R:142]: Let's keep the `version` variable.

File: DESCRIPTION [hunk 2]
```diff
@@ -20,7 +20,8 @@ Authors@R: c(
       comment = c(ORCID = "0000-0002-6714-8611")
     ),
     person("Vincent", "Fulco", role = "ctb"),
-    person("Alec", "Wong", role = "ctb")
+    person("Alec", "Wong", role = "ctb"),
+    person("Alex", "Brodersen", role = "ctb")
```
Comment (ijlyttle) [inline DESCRIPTION:24]: Not related to this line, but it's in the neighborhood. 

Just a heads up that just before merging the PR, we will manually bump the version based on where we will be at the time. No action needed by you, just so you will not be surprised.

File: NEWS.md [hunk 3]
```diff
@@ -11,6 +11,8 @@
 
 * uses `httr::RETRY()` for API requests to handle momentary issues with network connectivity. Thanks @jameslamb and @chircollab!
 
+* `box_ls()` provides the current `version_id` by default.
```
Comment (ijlyttle) [inline NEWS.md:14]: Can you add "(#181, @alexbrodersen)" after the end of the sentence?

**Review by ijlyttle — CHANGES_REQUESTED** [review]

**Comment by ijlyttle** [comment]
@alexbrodersen, I hit the wrong button before I could summarize the review. 

I don't think there is too much to do, once you are done, I'll be happy to bump the version.

As @nathancday points out, it will be good to have a test, but I had a look at the test and it's not too user friendly, so I will be happy to add a test before merging.

**Review by ijlyttle — APPROVED** [review]

**Comment by ijlyttle** [comment]
@alexbrodersen I'm doing some other work that will need to incorporate yours, so I hope it's OK that I made the requested changes.

@nathancday can you have a look to make sure what I've done is OK?

**Review by nathancday — COMMENTED** [review]
Returning `version_no` feels weird, the two identical columns. What about just `version` and `version_id`? Keep the legacy and add Alex's improvement for id

**Comment by ijlyttle** [comment]
The legacy was already split; `box_read()` and `box_dl()` both currently use a `version_no` argument.

I prefer `version_no` to `version` as it is more distinct from `version_id`. I agree it's awkward to return both, but I think we would have to do that for backwards compatibility (small as that risk might seem).

**Comment by nathancday** [comment]
Yea totally forgot about the existing split across args. Sounds good to have the weirdness of dupes for consistancy.

**Comment by ijlyttle** [comment]
Thanks @alexbrodersen!
